### ISSUE TITLE
Replace usage of Group-DETR with DN-DETR

### ISSUE THREAD

**Issue by Abdul-Mukit** [roboflow/rf-detr#390]
LW-DETR adopted [Group-DETR](https://arxiv.org/abs/2207.13085) instead of [DN-DETR](https://arxiv.org/abs/2203.01305).  
RF-DETR still uses it. Usage of 13 groups makes memory consumption very high. This becomes especially problematic for mask predictions. If we are using 13 groups, 200 queries, and 4 decoder layers, then we end up with  13 * 200 * (4 + 1 encoder_output) = 13,000 masks detected per image. This requires a high-end GPU to work with.    

Not everyone can afford high-end GPUs to train on. This also causes us to keep the batch size low. Lowering the batch size lowers the accuracy of each gradient step we are taking.  

However, there are alternatives like [DN-DETR](https://arxiv.org/abs/2203.01305).  
AFIK DN-DETR is much more memory efficient, and replacing Group-DETR with DN-DETR won't make too big of an impact when it comes to convergence.

<img width="782" height="608" alt="Image" src="https://github.com/user-attachments/assets/760052af-6fda-4731-8694-1d86b5fab40f" />  

<img width="821" height="555" alt="Image" src="https://github.com/user-attachments/assets/4b6ffbaa-a86b-4791-9389-a569c2950f90" />  
Images taken from [Group-DETR](https://arxiv.org/abs/2207.13085).  

My proposal is that maybe we should consider replacing Group-DETR with DN-DETR. This might make RFDetr more accessible.  
There are already examples in MaskDINO: https://github.com/IDEA-Research/MaskDINO/blob/3831d8514a3728535ace8d4ecc7d28044c42dd14/maskdino/modeling/transformer_decoder/maskdino_decoder.py#L191  

Are there any major limitations or problems in doing this? Or are there really any advantages of doing this?

**Comment by ShihuaHuang95** [comment]
@Abdul-Mukit In this case, you can have a try on our DEIMs (both v1 and v2), Dense O2O, no extra learnable parameters.

**Comment by Abdul-Mukit** [comment]
I am interested in working on this. Is this needed?

**Comment by isaacrob** [comment]
Worth noting that current training doesn't materialize most of the masks, so memory is significantly lower, like 3-5x for segmentation models. 

We don't really consider this necessary but may explore it in the future. Denoising uses the same memory as using two groups, and we don't find the memory improvement to be huge, especially because the memory of the decoder (essentially) doesn't scale with resolution. And as that graph shows, denoising gives worse results unless you do more complex methods / extra hyperparameter tuning. Our goal for using groups was to have as simple and as generalizable a method as possible without extra hyperparameter tuning :)

**Comment by isaacrob** [comment]
@ShihuaHuang95 I would love to see how well those models transfer to downstream tasks via RF100-VL! We didn't benchmark them ourselves because they seem to not meaningfully benefit from object detection pretraining on o365 based on your paper results and the lack of checkpoints other than X so we assumed they'd transfer worse than D-FINE.

**Comment by Abdul-Mukit** [comment]
Got it. Thank you.
